src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawWriter.cs

```diff
@@ -315,5 +417,25 @@ internal virtual void Close(WriteState currentState)
         {
             Close();
         }
+
+        private delegate bool GrowPrimitiveBufferDelegate<in T>(T value, Span<char> destination, out int charsWritten);
+        private void WriteWithBuffer<T>(T value, GrowPrimitiveBufferDelegate<T> checkFunc)
+        {
+            ArgumentNullException.ThrowIfNull(value);
+
+            //fast path without loop
+            if (checkFunc(value, _primitivesBuffer, out int charsWritten))
+            {
+                WriteChars(_primitivesBuffer, 0, charsWritten);
+                return;
+            }
+
+            while (!checkFunc(value, _primitivesBuffer, out charsWritten))
+            {
+                _primitivesBuffer = new char[_primitivesBuffer.Length * 2];
+            }
+
+            WriteChars(_primitivesBuffer, 0, charsWritten);
```

Comment:
I saw that there is difference between 
https://github.com/dotnet/runtime/blob/abb5c348f242e546b5768ad70ebe4051f08f7d24/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlUtf8RawTextWriter.cs#L560

and

https://github.com/dotnet/runtime/blob/abb5c348f242e546b5768ad70ebe4051f08f7d24/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlUtf8RawTextWriter.cs#L582

Which one should be used ?